faunadb-scala/src/test/scala/faunadb/CaseClassSpec.scala

```diff
@@ -0,0 +1,152 @@
+package faunadb
+
+import faunadb.values.Codec
+import faunadb.query._
+import org.scalatest.{ FlatSpec, Matchers }
+import scala.concurrent.ExecutionContext.Implicits.global
+
+class CaseClassSpec extends FlatSpec with Matchers with CommonSpec {
```

Comment:
I think there are pros and cons about this class.

On the bright site, I liked that we pulled these test cases out of the serialization and deserialization specs. It's a lot easier to understand the codec's behaviour with its effects being tested close together. However, the fact that we're trying to test Codecs without the usage of value types is confusing. In the way the test cases are written right now, they don't make sense if you think outside of the scope of this bug fix. My first reaction to this code is to question why they were not written using value types.

I still think there is value on keeping this class as long as we revert back to testing the Codec's conversion from value types to case classes and vice versa.

On that case, we go back to the need of having a test case that asserts their static independency from `import faunadb.values._`, which can be achieved by a separate test file with a case class containing all possible field types. I might no even need any specific assert since the file won't compile in case someone breaks the macro's implementation. However, we'd need some documentation at the top of the file to avoid someone deleting it for its apparent inutility.